torch/distributed/optim/zero_redundancy_optimizer.py

```diff
@@ -341,8 +368,7 @@ def __init__(
         self._partition_parameters_cache: List[List[Dict]] = []
         self._index_to_param_cache: List[torch.Tensor] = []
         self._device_to_params_per_rank_cache: Dict[torch.device, List[List[torch.Tensor]]] = {}
-        self._device_to_buckets_cache: Dict[torch.device, List[List[_DDPBucket]]] = {}
```

Comment:
I realized that there was no need to conform to this `Dict[torch.device, List[List[_DDPBucket]]]` mapping (namely, using `torch.device` as the key). I originally based it on `_device_to_params_per_rank`, which needs that kind of mapping due to the way `_partition_parameters()` assigns parameters to ranks.